tests/cxx/java_like/Account.cpp

```diff
@@ -22,6 +22,11 @@ void Account::transferTo(Account *target, int amount)
 //@ requires AccountPred(this, ?b) &*& AccountPred(target, ?t);
 //@ ensures AccountPred(this, b-amount) &*& AccountPred(target, t+amount);
 {
+    //@ open AccountPred(this, _);
+    //@ close AccountPred(this, _);
```

Comment:
To obtain `this != 0`

Reply:
It's more elegant to declare an autolemma
```
lemma_auto void AccountPred_inv();
    requires AccountPred(?account, ?balance);
    ensures AccountPred(account, balance) &*& account != 0;
```
It will be called whenever an `AccountPred` chunk is produced.

Reply:
(There will no longer be a need for this once we support instance predicates for C++. `myAccount->AccountPred(?balance)` implies `myAccount != 0`.)